### PR TITLE
Initial Wind River Linux support

### PR DESCRIPTION
**Pull request by radzy**
Signed-off-by: T.O. Radzy Radzykewycz <radzy@windriver.com>

### CODE REVIEW

File: cpe/openscap-cpe-oval.xml [hunk 1]
```diff
@@ -442,6 +442,20 @@
...
+                        <title>WR Linux 8</title>
+                        <affected family="unix">
+                            <platform>WR Linux 8</platform>
+                        </affected>
+                        <reference ref_id="cpe:/o:windriver:wrlinux:8" source="CPE"/>
+                        <description>The operating system installed on the system is WR Linux 8</description>
+                  </metadata>
+                  <criteria>
+                        <criterion comment="Installed operating system is part of the unix family" test_ref="oval:org.open-scap.cpe.wrlinux:tst:1"/>
+                        <!-- <criterion comment="WR Linux 8 is installed" test_ref="oval:org.open-scap.cpe.wrlinux:tst:8"/> -->
```
Comment (mpreisler) [inline cpe/openscap-cpe-oval.xml]: Why is this commented out?

As it is, all operating systems of unix family will behave as if they were WR Linux in openscap CPE.

File: src/OVAL/probes/unix/runlevel.c [hunk 2]
```diff
@@ -327,6 +332,15 @@ static int is_solaris (void)
         return (stat ("/etc/release", &st)   == 0);
 }
 
+static int is_wrlinux (void)
+{
+	struct stat st;
+	int retval;
+	if ( retval = stat ("/etc/os-release", &st) != 0 ) return(retval);
```
Comment (mpreisler) [inline src/OVAL/probes/unix/runlevel.c]: what is the logic behind this? WR Linux must not have /etc/os-release? Isn't the second line enough to make a good guess?

**Comment by mpreisler** [comment]
@openscap-jenkins add to whitelist

**Comment by radzy** [comment]
@mpreisler : The logic behind the /etc/os-release was that I wanted it to require /etc/os-release to be present.  This was for historical reasons, and I'm happy to remove it.  (Done in my local in-progress copy, so it will show up in the next upload.)  The check for /etc/wrlinux-release was added later (because we didn't already have /etc/wrlinux-release, so I had to create it.  But since I did that, os-release is no longer relevant.)

**Comment by radzy** [comment]
@mpreisler : The commented out test is an error.  I'm re-validating the latest version, with it in place.

**Comment by radzy** [comment]
@mpreisler : Incidentally, w.r.t. os-release/wrlinux-release, my internal reviewers suggest that I open and parse the /etc/wrlinux-release file to get the version number, instead of using the RPM metadata.  Nobody else does that.  Would this be an acceptable strategy ?

**Comment by mpreisler** [comment]
> @mpreisler : Incidentally, w.r.t. os-release/wrlinux-release, my internal reviewers suggest that I open and parse the /etc/wrlinux-release file to get the version number, instead of using the RPM metadata. Nobody else does that. Would this be an acceptable strategy ?

I am not sure how it works on WR Linux but on RHEL based distros we trust the rpm metadata more than contents of the file. However if you think it's preferable to parse the file you can. Check out the textfilecontent OVAL test.

**Comment by radzy** [comment]
Will upload a new PR when ready.

**Pull request closed without merging**